packages/frontend-2/components/server-management/DeleteInvitationDialog.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+  <LayoutDialog
+    v-model:open="isOpen"
+    max-width="sm"
+    :title="title"
+    :buttons="dialogButtons"
+  >
+    <div class="flex flex-col gap-6 text-sm text-foreground">
+      <p>Are you sure you want to delete the selected invitation?</p>
+      <div v-if="invite" class="flex flex-col gap-2">
+        <div class="flex gap-1">
+          <div class="w-20">Email:</div>
+          <strong>{{ invite.email }}</strong>
+        </div>
+        <div class="flex items-center gap-1">
+          <div class="w-20">Invited by:</div>
+          <UserAvatar :user="invite.invitedBy" />
+          {{ invite.invitedBy.name }}
+        </div>
+      </div>
+
+      <p>
+        This action
+        <strong>cannot</strong>
+        be undone.
+      </p>
+    </div>
+  </LayoutDialog>
+</template>
+
+<script setup lang="ts">
+import { useMutation } from '@vue/apollo-composable'
+import { LayoutDialog } from '@speckle/ui-components'
+import { InviteItem } from '~~/lib/server-management/helpers/types'
+import { adminDeleteInvite } from '~~/lib/server-management/graphql/mutations'
+import { useGlobalToast, ToastNotificationType } from '~~/lib/common/composables/toast'
+import {
+  ROOT_QUERY,
+  convertThrowIntoFetchResult,
+  getCacheId,
+  getFirstErrorMessage,
+  modifyObjectFields
+} from '~~/lib/common/helpers/graphql'
+import { AdminInviteList } from '~~/lib/common/generated/gql/graphql'
+
+const props = defineProps<{
+  open: boolean
+  title: string
+  invite: InviteItem | null
+}>()
+
+const emit = defineEmits<{
+  (e: 'update:open', val: boolean): void
+}>()
+
+const { triggerNotification } = useGlobalToast()
+const { mutate: adminDeleteMutation } = useMutation(adminDeleteInvite)
+
+const isOpen = computed({
+  get: () => props.open,
+  set: (newVal) => emit('update:open', newVal)
+})
+
+const deleteConfirmed = async () => {
+  const inviteId = props.invite?.id
+  if (!inviteId) {
+    return
+  }
+
+  const result = await adminDeleteMutation(
+    {
+      inviteId
+    },
+    {
+      update: (cache, { data }) => {
+        if (data?.inviteDelete) {
+          // Remove invite from cache
+          const cacheId = getCacheId('ServerInvite', inviteId)
+          cache.evict({
+            id: cacheId
+          })
+
+          // Modify 'admin' field of ROOT_QUERY so that we can modify all `inviteList` instances
+          modifyObjectFields<undefined, { [key: string]: AdminInviteList }>(
+            cache,
+            ROOT_QUERY,
+            (_fieldName, _variables, value, details) => {
+              // Find all `inviteList` fields (there can be multiple due to differing variables)
+              const inviteListFields = Object.keys(value).filter(
+                (k) =>
+                  details.revolveFieldNameAndVariables(k).fieldName === 'inviteList'
```

Comment:
This needs to be done cause `k` can be a value like `inviteList:{"query":"jon"}`, so the string contains the fieldName and variables serialized together. That's why you need `resolveFieldNameAndVariables` which will separate them and you can get just the field name on its own that way.